google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/StatementParser.java

```diff
@@ -457,17 +458,15 @@ static String removeStatementHint(String sql) {
     // searching for the first occurrence of a keyword that should be preceded by a closing curly
     // brace at the end of the statement hint.
     int startStatementHintIndex = sql.indexOf('{');
-    // Statement hints are only allowed for queries.
+    // Statement hints are allowed for both queries and DML statements.
     int startQueryIndex = -1;
     String upperCaseSql = sql.toUpperCase();
-    for (String keyword : selectStatements) {
+    Set<String> selectAndDmlStatements =
+        Sets.union(selectStatements, dmlStatements).immutableCopy();
```

Comment:
if style allows, consider moving the declaration and assignment on a single line.

Reply:
The line was split by the formatter so I guess this is what is needed to be style-conformant.